### ISSUE TITLE
"UndoRedo history mismatch: expected 0, got 1" when dragging image from FileSystem tab onto Visual Shader Editor

### ISSUE THREAD

**Issue by enxas** [godotengine/godot#79750]
### Godot version

v4.1.1.stable.mono.official [bd6af8e0e]

### System information

Godot v4.1.1.stable.mono - Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1660 Ti with Max-Q Design (NVIDIA; 31.0.15.3640) - Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 Threads)

### Issue description

Dragging and dropping image from FileSystem tab onto Visual Shader Editor produces error: `UndoRedo history mismatch: expected 0, got 1.`

### Steps to reproduce

1. Create MeshInstance3D.
2. Select that mesh and add Surface material Override -> New ShaderMaterial -> Visual Shader.
3. Drag and drop image from FileSystem tab onto Visual Shader Editor like shown in the video below

https://github.com/godotengine/godot/assets/8630984/c0e93cfe-2d2b-46b1-9aa0-a5f3ed095235

### Minimal reproduction project

NA

**Comment by KoBeWi** [comment]
Can't reproduce in 4.2 dev.

**Comment by enxas** [comment]
I have compiled Godot v4.2.dev.mono.custom_build [6588a4a29] and issue i still there. I have added MRP this time.
1. Double click `test.tres` in FileSystem to open Visual Shader Editor
2. From FileSystem click and drag `Grass002_1K_Color.jpg` onto Visual Shader Editor to create Texture2D node
3. After that error is printed `UndoRedo history mismatch: expected 0, got 3.` is printed

[MRP.zip](https://github.com/godotengine/godot/files/12135877/MRP.zip)
